### ISSUE TITLE
Crash while running on Linux x86_64 from release files

### ISSUE THREAD

**Issue by martinschwinzerl** [NearInfinityBrowser/NearInfinity#174]
When running the latest stable release on my current Debian stable installation, execution stops with this backtrace:
```
java.lang.NullPointerException
        at java.base/sun.nio.fs.UnixPath.register(UnixPath.java:881)
        at java.base/java.nio.file.Path.register(Path.java:894)
        at org.infinity.util.io.FileWatcher.register(Unknown Source)
        at org.infinity.util.io.FileWatcher.register(Unknown Source)
        at org.infinity.resource.Profile.lambda$initRootDirs$3(Unknown Source)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.infinity.resource.Profile.initRootDirs(Unknown Source)
        at org.infinity.resource.Profile.initGame(Unknown Source)
        at org.infinity.resource.Profile.init(Unknown Source)
        at org.infinity.resource.Profile.<init>(Unknown Source)
        at org.infinity.resource.Profile.openGame(Unknown Source)
        at org.infinity.NearInfinity$2.doInBackground(Unknown Source)
        at org.infinity.NearInfinity$2.doInBackground(Unknown Source)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "org.infinity.resource.Profile.getProperty(org.infinity.resource.Profile$Key)" is null
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at java.desktop/javax.swing.SwingWorker.get(SwingWorker.java:613)
        at org.infinity.NearInfinity.<init>(Unknown Source)
        at org.infinity.NearInfinity.main(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "org.infinity.resource.Profile.getProperty(org.infinity.resource.Profile$Key)" is null
        at org.infinity.gui.menu.SearchMenu.gameLoaded(Unknown Source)
        at org.infinity.gui.menu.BrowserMenuBar.gameLoaded(Unknown Source)
        at org.infinity.NearInfinity$2.doInBackground(Unknown Source)
        at org.infinity.NearInfinity$2.doInBackground(Unknown Source)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
```
If I run from the flatpak installation, it happens immediately. If I run using the jar file for the 2.4-20231231 release, the crash occurs after the selection of the .key file. The same occurs when I build locally against master branch and run the resulting jar file.  Build output from ant is as follows:

```
Buildfile: ~/downloads/git/NearInfinity/build.xml

clean:

compile:
    [mkdir] Created dir: /home/martin/downloads/git/NearInfinity/build/src
    [javac] Compiling 1116 source files to /home/martin/downloads/git/NearInfinity/build/src
    [javac] warning: [options] bootstrap class path not set in conjunction with -source 8
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] 1 warning
     [copy] Copying 219 files to /home/martin/downloads/git/NearInfinity/build/src
      [jar] Building jar: /home/martin/downloads/git/NearInfinity/NearInfinity.jar
```

Java jre/sdk versions are as follows:
```
java --version
openjdk 17.0.10 2024-01-16
OpenJDK Runtime Environment (build 17.0.10+7-Debian-1deb12u1)
OpenJDK 64-Bit Server VM (build 17.0.10+7-Debian-1deb12u1, mixed mode, sharing)
```
I did not test using the Oracle Java release yet, please let me know if you need any further information. 
Many thanks in advance!

**Comment by Argent77** [comment]
Thanks for the bug report.

Can you please build and run Near Infinity with additional debug information enabled? To do that, open `build.xml`, change the javac parameter `debug="false"` to `debug="true"`, and rebuild the app. (Alternatively download and run the [Nightly build](https://github.com/Argent77/NearInfinity/releases/tag/nightly) version). Please post the debug log output when the error occurs again.

To check whether it's caused by an incorrect app setting could you also navigate to `~/.java/.userPrefs/org` and temporarily rename the folder `infinity` to `infinity-1` (or something else), and restart Near Infinity? The app should ask for the game's chitin.key, followed by a first-time setup of the app preferences.

**Comment by martinschwinzerl** [comment]
Thank you for your quick response and for the instructions, much appreciated!

1) Rebuild with debug enabled in the build.xml file:
```
diff --git a/build.xml b/build.xml
index 0414892d..258a7cc6 100644
--- a/build.xml
+++ b/build.xml
@@ -29,7 +29,7 @@
            target="1.8"
            includeantruntime="false"
            deprecation="false"
-           debug="false"
+           debug="true"
            debuglevel="lines,vars,source"
            classpath="${classpath}:${jorbis.file}:${rsyntaxtextarea.file}:${jhexview.file}:${montemedia.file}:${apng_writer.file}:${json.file}:${commonmark.file}:${flatlaf.file}:${flatlafthemes.file}">
       <!-- Various debugging options: -->
```
results in more or less the same output as before:

```
java.lang.NullPointerException
        at java.base/sun.nio.fs.UnixPath.register(UnixPath.java:881)
        at java.base/java.nio.file.Path.register(Path.java:894)
        at org.infinity.util.io.FileWatcher.register(FileWatcher.java:195)
        at org.infinity.util.io.FileWatcher.register(FileWatcher.java:152)
        at org.infinity.resource.Profile.lambda$initRootDirs$3(Profile.java:2078)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.infinity.resource.Profile.initRootDirs(Profile.java:2077)
        at org.infinity.resource.Profile.initGame(Profile.java:1939)
        at org.infinity.resource.Profile.init(Profile.java:1767)
        at org.infinity.resource.Profile.<init>(Profile.java:1725)
        at org.infinity.resource.Profile.openGame(Profile.java:684)
        at org.infinity.NearInfinity$2.doInBackground(NearInfinity.java:440)
        at org.infinity.NearInfinity$2.doInBackground(NearInfinity.java:437)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "org.infinity.resource.Profile.getProperty(org.infinity.resource.Profile$Key)" is null
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at java.desktop/javax.swing.SwingWorker.get(SwingWorker.java:613)
        at org.infinity.NearInfinity.<init>(NearInfinity.java:471)
        at org.infinity.NearInfinity.main(NearInfinity.java:388)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "org.infinity.resource.Profile.getProperty(org.infinity.resource.Profile$Key)" is null
        at org.infinity.gui.menu.SearchMenu.gameLoaded(SearchMenu.java:88)
        at org.infinity.gui.menu.BrowserMenuBar.gameLoaded(BrowserMenuBar.java:176)
        at org.infinity.NearInfinity$2.doInBackground(NearInfinity.java:457)
        at org.infinity.NearInfinity$2.doInBackground(NearInfinity.java:437)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
```
I've attached the log of all messages occuring during build in case that is helpful. 

2) Renaming the app settings folder as suggested did not change anything, same error. 

3) I did also try to repeat the steps when switching to the `devel` branch - I assume that's more or less equivalent to running the nightlies?

I'm unfortunately very busy at the moment and just wanted to fire up NearInfinity to look something up. My java is rusty but I may be able to run this in Eclipse and set a breakpoint before the exception to better understand what's going on but it will take me a week or so to get to that - any pointers on how to help with the debugging (or any other tests that I can quickly run before that) would be great, thank you

**Comment by martinschwinzerl** [comment]
Sorry build log did not attach properly, here it is:
[build_log.txt](https://github.com/NearInfinityBrowser/NearInfinity/files/14549436/build_log.txt)

**Comment by Argent77** [comment]
Thank you, that was helpful.

Could you try out the attached version and see if it works? It should at least fix the NullPointer exception. I'm also interested in the error message that is printed to the debug window (if there is any).

[NearInfinity-watcher-patch-1.zip](https://github.com/NearInfinityBrowser/NearInfinity/files/14549648/NearInfinity-watcher-patch-1.zip)

**Comment by martinschwinzerl** [comment]
Thank you, the patched version works

**Comment by Argent77** [comment]
Interesting. It looks like you've hit a OS configuration limit.

I found this (slightly dated) topic which might help you solve the inotify watches limit on your system:
https://unix.stackexchange.com/questions/13751/kernel-inotify-watch-limit-reached

**Comment by Argent77** [comment]
A fix has already been committed (Argent77@cb0adf7bd976ae0c0c48eaa56e2d0323109f7695). Issue can be closed.
